### PR TITLE
MON-3814: Follow-up #2293

### PR DESCRIPTION
**Pull request by slashpai**
Follow-up from https://github.com/openshift/cluster-monitoring-operator/pull/2293

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -2106,6 +2106,8 @@ func (f *Factory) MetricsServerDeployment(apiAuthSecretName string, kubeletCABun
 		"--requestheader-extra-headers-prefix="+requestheaderExtraHeadersPrefix,
 		"--requestheader-group-headers="+requestheaderGroupHeaders,
 		"--requestheader-username-headers="+requestheaderUsernameHeaders,
+		"--tls-cert-file=/etc/client-ca-bundle/tls.crt",
```
Comment (machine424) [inline pkg/manifests/manifests.go:2109]: No test was failing when using `/etc/tls/private/xxx`, do they have the same content? Is testing this difficult?

Comment (slashpai) [reply pkg/manifests/manifests.go:2109]: no failure I just moved it from jsonnet to here 
https://github.com/openshift/cluster-monitoring-operator/pull/2293/files#r1557090512

Comment (slashpai) [reply pkg/manifests/manifests.go:2109]: https://github.com/openshift/cluster-monitoring-operator/blob/889bb2ce19cd1018888f506dd8c401cca163cfc8/jsonnet/components/metrics-server.libsonnet#L120 (metrics-server-tls) is been mounted at /etc/tls/private (https://github.com/openshift/cluster-monitoring-operator/blob/889bb2ce19cd1018888f506dd8c401cca163cfc8/jsonnet/components/metrics-server.libsonnet#L248)

In the updated approach as well it uses the same secret https://github.com/openshift/cluster-monitoring-operator/pull/2293/files#diff-0bbbc0b6e7ee4c4efe218030b64c2ef604fda58d92e51e52c206b6c5c0a4a5a3R158

Comment (slashpai) [reply pkg/manifests/manifests.go]: I updated the PR to remove unused in volume mounts https://github.com/openshift/cluster-monitoring-operator/pull/2293/files#r1557090512

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -2162,16 +2162,14 @@ func (f *Factory) MetricsServerSecret(tlsSecret *v1.Secret, apiAuthConfigmap *v1
 	var (
 		clientCA              = r.value("client-ca-file")
 		requestheaderClientCA = r.value("requestheader-client-ca-file")
-		tlsCA                 = r.value("tls.crt")
-		tlsKey                = r.value("tls.key")
 	)
```
[the remark this answers is not in the document]
Comment (rexagod) [reply pkg/manifests/manifests.go:2165]: True, and I believe the following can be dropped:

```go

		tlsSecret, err := t.client.WaitForSecretByNsName(ctx, types.NamespacedName{Namespace: t.namespace, Name: "metrics-server-tls"})
		if err != nil {
			return fmt.Errorf("failed to wait for metrics-server-tls secret: %w", err)
		}

```

[the remark this answers is not in the document]
Comment (machine424) [reply pkg/manifests/manifests.go:2165]: (i just realized we didin't need that wait. as we already waited for that secret here https://github.com/openshift/cluster-monitoring-operator/blob/65ee64665d9553c70f9cbfae5c2794d327c4fe32/pkg/tasks/metricsserver.go#L122-L128)

**Comment by slashpai** [comment]
Closing in favour of https://github.com/openshift/cluster-monitoring-operator/pull/2329

**Comment by machine424** [comment]
I'll
/reopen
this, it'll be nice to have this cleaning in 4.17

**Comment by machine424** [comment]
I think we can merge this once `tlsSecret` is dropped as an argument to `MetricsServerSecret`.

**Pull request closed without merging**